tools/crm_mon.c

```diff
@@ -788,12 +788,7 @@ detect_user_input(GIOChannel *channel, GIOCondition condition, gpointer user_dat
                 show ^= mon_show_bans;
                 break;
             case 'D':
-                /* If any header is shown, clear them all, otherwise set them all */
-                if (show & mon_show_summary) {
-                    show &= ~mon_show_summary;
-                } else {
-                    show |= mon_show_summary;
-                }
+                show ^= mon_show_summary;
```

Comment:
Not quite the same :) The interactive input can set the individual components of the summary on or off. 'D' will clear all of them or set all of them, whereas ^= will flip each individually.